server/src/main/java/org/opensearch/cluster/routing/allocation/AwarenessReplicaBalance.java

```diff
@@ -101,12 +101,21 @@ public int maxAwarenessAttributes() {
         return awarenessAttributes;
     }
 
-    public Optional<String> validate(int replicaCount) {
-        if ((replicaCount + 1) % maxAwarenessAttributes() != 0) {
-            String errorMessage = "expected total copies needs to be a multiple of total awareness attributes ["
-                + maxAwarenessAttributes()
-                + "]";
-            return Optional.of(errorMessage);
+    public Optional<String> validate(int replicaCount, int autoExpandMaxCap) {
+        if (autoExpandMaxCap == -1) {
```

Comment:
`autoExpandMaxCap` will be `Integer.MAX_VALUE` when auto expand is set to all . We will need to ignore checks in those cases as well .